libs/cypress/src/lib/eip1193-bridge.ts

```diff
@@ -0,0 +1,121 @@
+      method = args[0].method;
+      params = args[0].params;
+    } else {
+      method = args[0];
+      params = args[1];
+    }
+    try {
+      // Mock out request accounts and chainId
+      if (method === 'eth_requestAccounts' || method === 'eth_accounts') {
+        const address = this.signer ? [await this.signer.getAddress()] : [];
+        if (isCallbackForm) {
+          callback({ result: address });
+        } else {
+          return Promise.resolve(address);
+        }
+      }
+      if (method === 'eth_chainId') {
+        if (isCallbackForm) {
+          callback(null, { result: '0x3' });
+        } else {
+          return Promise.resolve('0x3');
+        }
+      }
+      // Hacky, https://github.com/ethers-io/ethers.js/issues/1683#issuecomment-1016227588
+
+      // If from is present on eth_call it errors, removing it makes the library set
+      // from as the connected wallet which works fine
+      if (params && params.length && params[0].from && method === 'eth_call')
+        delete params[0].from;
+      let result;
+      // For sending a transaction if we call send it will error
+      // as it wants gasLimit in sendTransaction but hexlify sets the property gas
+      // to gasLimit which makes sensd transaction error.
+      // This has taken the code from the super method for sendTransaction and altered
+      // it slightly to make it work with the gas limit issues.
+      if (
+        params &&
+        params.length &&
+        params[0].from &&
+        method === 'eth_sendTransaction'
+      ) {
+        // Hexlify will not take gas, must be gasLimit, set this property to be gasLimit
+        params[0].gasLimit = params[0].gas;
+        delete params[0].gas;
+        // If from is present on eth_sendTransaction it errors, removing it makes the library set
+        // from as the connected wallet which works fine
+        delete params[0].from;
+        const req = ethers.providers.JsonRpcProvider.hexlifyTransaction(
+          params[0]
+        );
+        // Hexlify sets the gasLimit property to be gas again and send transaction requires gasLimit
+        req['gasLimit'] = req['gas'];
+        delete req['gas'];
+
+        if (!this.signer) {
+          throw new Error('No signer');
+        }
+
+        // Send the transaction
+        const tx = await this.signer.sendTransaction(req);
+        result = tx.hash;
+      } else {
+        // All other transactions the base class works for
+        result = await super.send(method, params);
+      }
+      console.debug('result received', method, params, result);
+      if (isCallbackForm) {
+        callback(null, { result });
+      } else {
+        return result;
+      }
+    } catch (error) {
+      console.log(error);
+      if (isCallbackForm) {
+        callback(error, null);
+      } else {
+        throw error;
+      }
+    }
+  }
+}
+
+const MNEMONIC =
+  'first length secret attitude green cloth aspect lucky false seat soap seat';
```

Comment:
chore (blocking): we will need a new wallet with this as a secret set by an env var now it has been committed to source. It should also be added to our password manager for future people.

Reply:
can you set this in github secrets as `wallet_mnemonic` and we can reference this in the test process.env.wallet_mnemonic

Reply:
This exists as `CYPESS_ETH_WALLET_MNEMONIC ` you can add this using environment variables